File: src/backend/backend.Core/Models/Lesson.cs
namespace backend.Core.Models;

public class Lesson
{
    public string Code { get; set; }
    
    public string Title { get; set; }
    
    public string Content { get; set; }
    
    public string Author { get; set; }
    
    public List<string> Supplements { get; set; }
    
    public List<Trait> Traits { get; set; }
    
    public List<Task> Tasks { get; set; }
    
    public List<Topic> Topics { get; set; }
}

File: src/frontend/src/entities/Lesson.ts
import { SortAndFilterRequest, getSortAndFilterRequestPayload } from '~/shared/api'
import { apiClient } from '~/shared/apiClient'

export interface Lesson {
  code: string
  title: string
  content: string
  traits: string[]
  supplements: string[]
  tasks: string[]
  author: string
}

export const LessonsApi = {
  async getAll(payload: SortAndFilterRequest = getSortAndFilterRequestPayload()) {
    const { data } = await apiClient.post<Lesson[]>('/OM/Lessons', payload)

    return data
  },
  async getOne(code: Lesson['code']) {
    const { data } = await apiClient.post<Lesson>('/OM/Lesson', { code })

    return data
  },
}
